### DISCUSSION TITLE
Protractor to WDIO Migration

### DISCUSSION THREAD

**Discussion by rgprakash** [webdriverio/codemod#26] [Migration Issue]
Hi team,
I am trying to migrate a protractor .js file which has set of reusable functions (click, setText etc.,). These functions will check for the state of element and then click or enter text. Sample code is attached.

When i try to migrate, i get the error  **ERR ./Base/CommonFunctions.ts Transformation error (Unexpected token (3:12))
SyntaxError: Unexpected token (3:12)**
I
[CommonFunctions.zip](https://github.com/webdriverio/codemod/files/6546145/CommonFunctions.zip)

How to handle these Protractor ExpectedConditions in WebDriverIO migration?

**Comment by DMazona** [comment]
If you're trying to convert a ts file (TypeScript) try the --parser=ts option on the command line.
